TestParamRef/ClusterScopedParamRefWithNameDenyNotFound flaking

### Which jobs are flaking?

So far, only https://prow.k8s.io/job-history/gs/kubernetes-ci-logs/pr-logs/directory/pull-kubernetes-unit. But I have seen it fail twice in two different PRs today.

### Which tests are flaking?

k8s.io/apiserver/pkg/admission/plugin/policy/validating, TestParamRef/ClusterScopedParamRefWithNameDenyNotFound

```
=== RUN   TestParamRef/ClusterScopedParamRefWithNameDenyNotFound
=== PAUSE TestParamRef/ClusterScopedParamRefWithNameDenyNotFound
=== CONT  TestParamRef/ClusterScopedParamRefWithNameDenyNotFound
    admission_test.go:1783: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1783
        	            				/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1521
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestParamRef/ClusterScopedParamRefWithNameDenyNotFound
--- FAIL: TestParamRef/ClusterScopedParamRefWithNameDenyNotFound (4.23s)
```


### Since when has it been flaking?

Around 2025-05-30?


### Testgrid link

_No response_

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig api-machinery

Update - [Triage](https://storage.googleapis.com/k8s-triage/index.html?job=ci-kubernetes-unit&test=k8s.io%2Fapiserver%2Fpkg%2Fadmission%2Fplugin%2Fpolicy%2Fvalidating.validating&xjob=ci-kubernetes-unit-)
- Observed 7 failures grouped into 4 clusters
---
- TestParamRef/ClusterScopedParamRefWithNameDenyNotFound [Prow](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit/1933148584010584064)
```
{Failed  === RUN   TestParamRef/ClusterScopedParamRefWithNameDenyNotFound
=== PAUSE TestParamRef/ClusterScopedParamRefWithNameDenyNotFound
=== CONT  TestParamRef/ClusterScopedParamRefWithNameDenyNotFound
    admission_test.go:1783: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1783
        	            				/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1521
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestParamRef/ClusterScopedParamRefWithNameDenyNotFound
--- FAIL: TestParamRef/ClusterScopedParamRefWithNameDenyNotFound (4.24s)

=== RUN   TestParamRef
--- FAIL: TestParamRef (0.00s)
}
```
---
- TestParamRef/ParamRefWithNameDenyNotFound [Prow](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit/1932769079681093632)
```
{Failed  === RUN   TestParamRef/ParamRefWithNameDenyNotFound
=== PAUSE TestParamRef/ParamRefWithNameDenyNotFound
=== CONT  TestParamRef/ParamRefWithNameDenyNotFound
    admission_test.go:1783: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1783
        	            				/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1521
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestParamRef/ParamRefWithNameDenyNotFound
--- FAIL: TestParamRef/ParamRefWithNameDenyNotFound (4.25s)

=== RUN   TestParamRef
--- FAIL: TestParamRef (0.00s)
}
```
---
- TestParamRef/ParamRefWithLabelSelectorWithNamespaceDenyNotFound [Prow](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit/1931145340145111040)
```
{Failed  === RUN   TestParamRef/ParamRefWithLabelSelectorWithNamespaceDenyNotFound
=== PAUSE TestParamRef/ParamRefWithLabelSelectorWithNamespaceDenyNotFound
=== CONT  TestParamRef/ParamRefWithLabelSelectorWithNamespaceDenyNotFound
    admission_test.go:1783: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1783
        	            				/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1521
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestParamRef/ParamRefWithLabelSelectorWithNamespaceDenyNotFound
--- FAIL: TestParamRef/ParamRefWithLabelSelectorWithNamespaceDenyNotFound (4.29s)

=== RUN   TestParamRef
--- FAIL: TestParamRef (0.00s)
}
```
---
- TestParamRef/ParamRefWithEverythingSelectorWithNamespaceDenyNotFound [Prow](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit/1931857564018937856)
```
{Failed  === RUN   TestParamRef/ParamRefWithEverythingSelectorWithNamespaceDenyNotFound
=== PAUSE TestParamRef/ParamRefWithEverythingSelectorWithNamespaceDenyNotFound
=== CONT  TestParamRef/ParamRefWithEverythingSelectorWithNamespaceDenyNotFound
    admission_test.go:1783: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1783
        	            				/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1521
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestParamRef/ParamRefWithEverythingSelectorWithNamespaceDenyNotFound
--- FAIL: TestParamRef/ParamRefWithEverythingSelectorWithNamespaceDenyNotFound (4.25s)

=== RUN   TestParamRef
--- FAIL: TestParamRef (0.00s)
}
```
---
- TestParamRef/ParamRefWithLabelSelectorWithNamespaceAllowNotFound [Prow](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit/1931205738374369280)
```
{Failed  === RUN   TestParamRef/ParamRefWithLabelSelectorWithNamespaceAllowNotFound
=== PAUSE TestParamRef/ParamRefWithLabelSelectorWithNamespaceAllowNotFound
=== CONT  TestParamRef/ParamRefWithLabelSelectorWithNamespaceAllowNotFound
    admission_test.go:1783: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1783
        	            				/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1521
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestParamRef/ParamRefWithLabelSelectorWithNamespaceAllowNotFound
--- FAIL: TestParamRef/ParamRefWithLabelSelectorWithNamespaceAllowNotFound (4.31s)

=== RUN   TestParamRef
--- FAIL: TestParamRef (0.00s)
}
```
---
- TestParamRef/ClusterScopedParamRefWithEverythingSelectorAllowNotFound [Prow](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit/1929824372353142784)
```
{Failed  === RUN   TestParamRef/ClusterScopedParamRefWithEverythingSelectorAllowNotFound
=== PAUSE TestParamRef/ClusterScopedParamRefWithEverythingSelectorAllowNotFound
=== CONT  TestParamRef/ClusterScopedParamRefWithEverythingSelectorAllowNotFound
    admission_test.go:1783: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1783
        	            				/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go:1521
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestParamRef/ClusterScopedParamRefWithEverythingSelectorAllowNotFound
--- FAIL: TestParamRef/ClusterScopedParamRefWithEverythingSelectorAllowNotFound (4.17s)

=== RUN   TestParamRef
--- FAIL: TestParamRef (0.00s)
}
```

--- a/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/admission_test.go
@@ -360,6 +360,7 @@ func setupTestCommon(
 	matcher generic.PolicyMatcher,
 	shouldStartInformers bool,
 ) *generic.PolicyTestContext[*validating.Policy, *validating.PolicyBinding, validating.Validator] {
+	t.Cleanup(generic.SetPolicyRefreshIntervalForTests(10 * time.Millisecond))
 	testContext, testContextCancel, err := generic.NewPolicyTestContext(
 		t,
 		validating.NewValidatingAdmissionPolicyAccessor,
@@ -1508,7 +1509,6 @@ func TestParamRef(t *testing.T) {
 						}
 
 						t.Run(name, func(t *testing.T) {
-							t.Parallel()
 							// Test creating a policy with a cluster or namesapce-scoped param
 							// and binding with the provided configuration. Test will ensure
 							// that the provided configuration is capable of matching

